pkg/bitmap/bitmap.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bitmap provides the implementation of bitmap.
+package bitmap
+
+import (
+	"math"
+	"math/bits"
+)
+
+// Bitmap struct contains numOnes which represent number of ones in the
+// bitmap and a slice bitBlock which represent bits. The type of bitBlock
+// is uint64 which means each number in bitBlock can contain information
+// for 64 bits.
+//
+// +stateify savable
+type Bitmap struct {
+	numOnes  uint32
+	bitBlock []uint64
+}
```

Comment:
I think it makes more sense to put comments on the individual fields. Something like:
```go
// Bitmap implements an efficient bitmap.
//
// +stateify savable
type Bitmap struct {
	// numOnes is the number of ones in the bitmap.
	numOnes  uint32

	// bitBlock holds the bits. The type of bitBlock is uint64 which means
	// each number in bitBlock contains 64 entries.
	bitBlock []uint64
}
```